From f4bdf8bdec0af294bc437286636d60781e192e18 Mon Sep 17 00:00:00 2001
From: jinghua <jinghua@marvell.com>
Date: Thu, 21 Jan 2016 17:07:39 +0800
Subject: [PATCH 0740/1240] fdt: a3700: update pre-defined presets for DB board
 configuration

	there are two main configurations for Armada3700 DB, between
	these two presets, most of the configurations are the same:
	* CPU 600MHz and DDR 600MHz
	* SATA
	* neta0 - RGMII-0
	* PCIe(COMPHY-0)

	the only difference are COMPHY-1 and RGMII-1
	preset 1: COMPHY-1 is working as USB3 and neta1 with RGMII-1
	preset 2. COMPHY-1 is working as SGMII-1 and no RGMII-1

	In this patch,
	- modify arch/arm/dts/armada-lp-db.dts to the 1stpreset,
	  which is COMPHY-1 is working as USB3
	- create arch/arm/dts/armada-lp-db-sgmii1.dts to the 2nd preset.
	  which is COMPHY-1 is working as SGMII-1

Change-Id: I9138508f5f2d3822916599486c636f7724963377
Signed-off-by: jinghua <jinghua@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/26957
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
---
 arch/arm/dts/Makefile                |  1 +
 arch/arm/dts/armada-lp-db-sgmii1.dts | 41 ++++++++++++++++++++++++++++++++++++
 arch/arm/dts/armada-lp-db.dts        | 25 +++++++++++++++++++---
 3 files changed, 64 insertions(+), 3 deletions(-)
 create mode 100644 arch/arm/dts/armada-lp-db-sgmii1.dts

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 0c7c11c..3a6fd43 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -54,6 +54,7 @@ dtb-$(CONFIG_SOCFPGA) += socfpga_cyclone5_socrates.dtb
 dtb-$(CONFIG_TARGET_ARMADA_38X) += armada-385-db.dtb
 dtb-$(CONFIG_TARGET_ARMADA_LP) += armada-lp-db.dtb \
 				  armada-lp-palladium.dtb \
+				  armada-lp-db-sgmii1.dtb \
 				  armada-lp-customer0.dtb
 dtb-$(CONFIG_TARGET_ARMADA_8K) += apn-806-db.dtb \
 		       apn-806-pd.dtb \
diff --git a/arch/arm/dts/armada-lp-db-sgmii1.dts b/arch/arm/dts/armada-lp-db-sgmii1.dts
new file mode 100644
index 0000000..115ea97
--- /dev/null
+++ b/arch/arm/dts/armada-lp-db-sgmii1.dts
@@ -0,0 +1,41 @@
+
+#include "armada-lp-db.dts"
+
+/* detail board setup:
+ * Pcie Jumper slot	:	DB-88F3720-PCIe-mPCIe-Jumper
+ * SGMII module slot	:	DB-88F3720-SerDes-Jumper
+ * RGMII module slot	:	DB-88F3720-PHY module
+ * eMMC slot		:	DB-88F3720-eMMC_NAND module
+ * SDIO slot		:	DB-88F3720-SDcard module
+ *
+ * PHY0 (USB3/SGMII1)	:	SGMII1_Ethernet via PHY (1 Lane)
+ * PHY1 (PCIe/SGMII0)	:	PCIe (WiFi via mini_PCIe module)
+ * PHY2 (SATA)		:	SATA
+ * RGMII		:	Ethernet via PHY (1 Lane)
+ * USB2 Port 0		:	USB2 only
+ * USB2 Port 1		:	USB2 Host
+ * eMMC*		:	eMMC NAND
+ * SDIO*		:	SD card
+ * BOOT device*		:	SPI Flash
+*/
+
+/ {
+	model = "DB-88F3720-DDR3-Modular-SGMII1";
+	compatible = "marvell,armada-lp-db0", "marvell,armada-lp";
+
+	soc {
+		internal-regs {
+			neta1: neta@40000 {
+				status = "okay";
+				phy_addr = <0x1>;
+				phy_mode = "sgmii";
+			};
+			comphy {
+				phy1 {
+					phy-type = <PHY_TYPE_SGMII1>;
+					phy-speed = <PHY_SPEED_1_25G>;
+				};
+			};
+		};
+	};
+};
diff --git a/arch/arm/dts/armada-lp-db.dts b/arch/arm/dts/armada-lp-db.dts
index c0d0413..51cbf84 100644
--- a/arch/arm/dts/armada-lp-db.dts
+++ b/arch/arm/dts/armada-lp-db.dts
@@ -3,6 +3,25 @@
 #include <dt-bindings/ddr/ddr_data.h>
 #include "armada-lp.dtsi"
 
+/* detail board setup:
+ * Pcie Jumper slot	:	DB-88F3720-PCIe-mPCIe-Jumper
+ * SGMII module slot	:	DB-88F3720-SerDes-Jumper
+ * RGMII module slot	:	DB-88F3720-PHY module
+ * eMMC slot		:	DB-88F3720-eMMC_NAND module
+ * SDIO slot		:	DB-88F3720-SDcard module
+ *
+ * PHY0 (USB3/SGMII1)	:	USB3
+ * PHY1 (PCIe/SGMII0)	:	PCIe (WiFi via mini_PCIe module)
+ * PHY2 (SATA)		:	SATA
+ * RGMII-0		:	Ethernet via PHY (1 Lane)
+ * RGMII-1		:	Ethernet via PHY (1 Lane)
+ * USB2 Port 0		:	USB2 as part USB3
+ * USB2 Port 1		:	USB2 Host
+ * eMMC*		:	eMMC NAND
+ * SDIO*		:	SD card
+ * BOOT device*		:	SPI Flash
+*/
+
 / {
 	model = "DB-88F3720-DDR3-Modular";
 	compatible = "marvell,armada-lp-db0", "marvell,armada-lp";
@@ -34,7 +53,7 @@
 			neta1: neta@40000 {
 				status = "okay";
 				phy_addr = <0x1>;
-				phy_mode = "sgmii";
+				phy_mode = "rgmii";
 			};
 			i2c: i2c@11000 {
 				status = "okay";
@@ -45,8 +64,8 @@
 					phy-speed = <PHY_SPEED_2_5G>;
 				};
 				phy1 {
-					phy-type = <PHY_TYPE_SGMII1>;
-					phy-speed = <PHY_SPEED_1_25G>;
+					phy-type = <PHY_TYPE_USB3_HOST0>;
+					phy-speed = <PHY_SPEED_5G>;
 				};
 			};
 			pcie-controller {
-- 
1.9.1

